rs/nns/integration_tests/src/cycles_minting_canister.rs

```diff
@@ -1755,6 +1766,79 @@ fn cmc_notify_top_up_not_rate_limited_by_invalid_top_up() {
     assert_eq!(cycles, Cycles::new(400_000_000_000_000u128));
 }
 
+#[test]
+fn test_cmc_topups_from_subnet_rental_canister_do_not_affect_limit() {
+    let state_machine = state_machine_builder_for_nns_tests().build();
+
+    let account = AccountIdentifier::new(*TEST_USER1_PRINCIPAL, None);
+    let src_account = AccountIdentifier::new(SUBNET_RENTAL_CANISTER_ID.get(), None);
+    // The only requirement here is to have sufficient funds. Other than that,
+    // the precise number here does not matter.
+    let balance = Tokens::new(1e6 as u64, 0).unwrap();
+    let nns_init_payloads = NnsInitPayloadsBuilder::new()
+        .with_test_neurons()
+        .with_ledger_account(src_account, balance)
+        .with_ledger_account(account, balance)
+        .build();
+    setup_nns_canisters(&state_machine, nns_init_payloads);
+
+    // Conversion rate in tests is 100 XDR per ICP
+    let tokens = CYCLES_MINTING_LIMIT as u64 / ONE_TRILLION / 100;
```

Comment:
Despite your comment, I do not know where this formula is coming from. Giving this variable a more elaborate name would probably help.

Reply:
After reading the rest of this test, I think I understand what this quantity is: this converts to more than half the number of cycles allowed in a single burst. Name should indicate.

Reply:
I updated the comment, hopefully clearer: 
```
    // Conversion rate in tests is 100 XDR per ICP, and cycles cost 1 XDR per trillion cycles.
    // To get the ICP amount for maximum cycles, we divide by 1 trillion and then by 100.
 ```